atc/creds/vault/vault.go

```diff
@@ -59,10 +65,31 @@ func (v Vault) Get(varDef template.VariableDefinition) (interface{}, bool, error
 	return evenLessTyped, true, nil
 }
 
+func (v Vault) getSecretReader() (SecretReader, error) {
+	v.once.Do(func() {
```

Comment:
At first I thought the `v.once` was unnecessary but it actually probably helps to not construct the `time.After`. 👍